src/utils/focusTrapComponent.ts

```diff
@@ -0,0 +1,72 @@
+import { FocusTrap as _FocusTrap, Options as FocusTrapOptions, createFocusTrap } from "focus-trap";
+import { FocusableElement, focusElement } from "./dom";
+
+/**
+ * Defines interface for components with a focus trap.
+ */
+export interface FocusTrapComponent {
+  /**
+   * The focus trap instance.
+   */
+  focusTrap: FocusTrap;
+
+  /**
+   * The focus trap element.
+   */
+  focusTrapEl: HTMLElement;
+}
+
+export type FocusTrap = _FocusTrap;
+
+/**
+ * Helper to set up focus trap component.
+ *
+ * @param component
+ */
+export function connectFocusTrap(component: FocusTrapComponent): void {
+  const { focusTrapEl } = component;
+
+  if (!focusTrapEl) {
+    return;
+  }
+
+  if (focusTrapEl.tabIndex == null) {
+    focusTrapEl.tabIndex = -1;
+  }
+
+  const focusTrapOptions: FocusTrapOptions = {
+    allowOutsideClick: true,
```

Comment:
Do we need to set the defaults on `delayInitialFocus`, `clickOutsideDeactivates`, `returnFocusOnDeactivate`?

Reply:
Nope. i'll remove :)